src/length.rs

```diff
@@ -33,6 +33,7 @@ use std::fmt;
 /// another. See the `ScaleFactor` docs for an example.
 // Uncomment the derive, and remove the macro call, once heapsize gets
 // PhantomData<T> support.
+#[repr(C)]
 #[derive(RustcDecodable, RustcEncodable)]
 pub struct Length<T, Unit>(pub T, PhantomData<Unit>);
```

Comment:
Note that this is safe, even though rustc will complain about it until https://github.com/rust-lang/rust/pull/39462 lands.

Also, note that depending of the size of T, being Length a struct but T an integer/floating point value, passing `Length` as an argument instead of `T` may break the ABI in 32bit linux, etc, see the last comments in https://github.com/rust-lang/rfcs/pull/1758.

I don't think you're going to pass Length<T> directly through, right?